mail_tracking/models/ir_mail_server.py

```diff
@@ -49,9 +48,16 @@ def build_email(self, email_from, email_to, subject, body, email_cc=None,
         return msg
 
     def _tracking_email_get(self, message):
-        tracking_email_id = False
-        if message.get('X-Odoo-Tracking-ID', '').isdigit():
-            tracking_email_id = int(message['X-Odoo-Tracking-ID'])
+        try:
+            tracking_email_id = int(
+                message.get(
+                    "X-Odoo-MailTracking-ID",
+                    # Deprecated tracking header, kept as fallback
+                    message["X-Odoo-Tracking-ID"],
+                )
+            )
+        except (TypeError, ValueError, KeyError):
+            tracking_email_id = False
```

Comment:
`try`/`catch` is more expensive than a simple `if` (at least up to Python 3.11)
```suggestion
        mt_obj = self.env["mail.tracking.email"]
        tracking_email_id = message.get(
            "X-Odoo-MailTracking-ID",
            # Deprecated tracking header, kept as fallback
            message.get("X-Odoo-Tracking-ID", "")
        )
        if not tracking_email_id.isdigit():
            return mt_obj
        return mt_obj.browse(int(tracking_email_id))
```